Ekubo wETH-ETH liquidity pool.ipynb

```diff
@@ -0,0 +1,1405 @@
+    "median_prices_df['price_plus_20%'] = median_prices_df['median_price'] * (1 + (price_range_liquidity*0.01))\n",
+    "median_prices_df['tick_id_median'] = median_prices_df['median_price'].apply(price_to_tick_id)\n",
+    "median_prices_df['tick_id_minus_20%'] = median_prices_df['price_minus_20%'].apply(price_to_tick_id)\n",
+    "median_prices_df['tick_id_plus_20%'] = median_prices_df['price_plus_20%'].apply(price_to_tick_id)\n",
+    "\n",
+    "# Step 6: Round the tick_ids to the nearest multiple of tick_spacing_touse\n",
+    "tick_spacing_touse = 20  # Define tick_spacing\n",
+    "median_prices_df['tick_id_minus_20%_rounded'] = median_prices_df['tick_id_minus_20%'].apply(lambda x: round_to_tick_spacing(x, tick_spacing_touse))\n",
+    "median_prices_df['tick_id_plus_20%_rounded'] = median_prices_df['tick_id_plus_20%'].apply(lambda x: round_to_tick_spacing(x, tick_spacing_touse))\n",
+    "\n",
+    "# Step 7: Create an array of tick IDs for each block\n",
+    "median_prices_df['tick_id_array'] = median_prices_df.apply(\n",
+    "    lambda row: np.arange(row['tick_id_minus_20%_rounded'], row['tick_id_plus_20%_rounded'] + tick_spacing_touse, tick_spacing_touse), axis=1\n",
+    ")\n",
+    "\n",
+    "# Step 8: Retain the necessary columns\n",
+    "final_df = median_prices_df[['block_number', 'median_price', 'tick_id_median', 'tick_id_array']]\n",
+    "\n",
+    "# Step 9: Distribute cumulative liquidity per block\n",
+    "cumulative_liquidity_per_tick = {}\n",
+    "\n",
+    "liquidity_per_block = []  # To store cumulative liquidity for each block\n",
+    "\n",
+    "for idx, row in final_df.iterrows():\n",
+    "    block_number = row['block_number']\n",
+    "    tick_array = row['tick_id_array']\n",
+    "\n",
+    "    # Initialize cumulative liquidity if not already initialized\n",
+    "    if not cumulative_liquidity_per_tick:\n",
+    "        cumulative_liquidity_per_tick = {tick: 0 for tick in tick_array}\n",
+    "\n",
+    "    # Clone the current cumulative state so that we can update it for the current block\n",
+    "    current_block_liquidity = cumulative_liquidity_per_tick.copy()\n",
+    "\n",
+    "    # Get liquidity data for the current block only\n",
+    "    liquidity_data_for_block = [event for event in ekubo_eth_weth_liquidity_data if event['block_number'] == block_number]\n",
+    "    \n",
+    "    # Add the current block's liquidity to the cumulative state\n",
+    "    for liquidity_event in liquidity_data_for_block:\n",
+    "        tick_lower = liquidity_event['tick_lower']\n",
+    "        tick_upper = liquidity_event['tick_upper']\n",
+    "        delta_liquidity = liquidity_event['delta_liquidity']\n",
+    "        \n",
+    "        # Distribute liquidity to the relevant ticks in the cumulative state\n",
+    "        for tick in tick_array:\n",
+    "            if tick_lower <= tick <= tick_upper:\n",
```

Comment:
i think it has to be tick < tick_upper (as per uniswap v3 math, ekubo could be different. u need to check(